File: src/main/java/com/dotronglong/faker/controller/MainController.java
package com.dotronglong.faker.controller;

import com.dotronglong.faker.contract.Handler;
import com.dotronglong.faker.contract.Router;
import com.dotronglong.faker.service.handler.NotFoundHandler;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.util.Objects;

@RestController
public class MainController {
    private Router router;

    @Autowired
    public MainController(Router router) {
        this.router = router;
    }

    @RequestMapping(value = "/**/{[path:[^\\.]*}")
    public void index(final HttpServletRequest request, final HttpServletResponse response) {
        Handler handler = router.match(request);
        if (Objects.isNull(handler)) {
            handler = new NotFoundHandler();
        }

        handler.handle(request, response);
    }
}

File: src/main/kotlin/com/dotronglong/faker/controller/MainController.kt
package com.dotronglong.faker.controller

import com.dotronglong.faker.contract.Router
import com.dotronglong.faker.service.handler.NotFoundHandler
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.server.reactive.ServerHttpRequest
import org.springframework.http.server.reactive.ServerHttpResponse
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
class MainController @Autowired constructor(val router: Router) {

    @CrossOrigin(
            origins = ["*"],
            methods = [
                RequestMethod.GET,
                RequestMethod.POST,
                RequestMethod.PUT,
                RequestMethod.PATCH,
                RequestMethod.DELETE,
                RequestMethod.OPTIONS
            ],
            exposedHeaders = [
                "Authorization",
                "Access-Control-Allow-Origin",
                "Access-Control-Allow-Credentials"
            ],
            maxAge = 86400L,
            allowCredentials = "true"
    )
    @RequestMapping(
            path = ["/**"],
            method = [
                RequestMethod.GET,
                RequestMethod.POST,
                RequestMethod.PUT,
                RequestMethod.PATCH,
                RequestMethod.DELETE,
                RequestMethod.OPTIONS
            ]
    )
    fun handle(request: ServerHttpRequest, response: ServerHttpResponse): Mono<Void> {
        val handler = router.match(request) ?: NotFoundHandler()

        return handler.handle(request, response)
    }
}